covid_graphs/covid_graphs/formula.py

```diff
@@ -0,0 +1,64 @@
+import math
+from datetime import datetime, timedelta
+import numpy as np
+from collections import namedtuple
+from enum import Enum
+
+EXPONENT = 6.23
+
+# TODO(lukas): Move to dataclass?
+Formula = namedtuple('Formula', ['lambd', 'text', 'second_ip_day', 'min_case_count'])
+
+
+class XAxisType(Enum):
+    Dated = 'dated'  # with dates
+    Numbered = 'numbered'  # numbered instead of dates
+
+    def __str__(self):
+        return self.value
+
+
+def ATG_formula(TG, A, exponent=EXPONENT, min_case_count=200):
+    text = r'$\frac{_A}{_TG} \cdot \left(\frac{t}{_TG}\right)^{_expon} / e^{t/_TG}$'
+    text = text.replace("_A", f"{A}").replace("_TG", f"{TG}").replace("_expon", f"{exponent}")
+    # Second inflection point day
+    second_ip_day = math.ceil(TG * (exponent + math.sqrt(exponent)))
+    return Formula(lambda t: (A / TG) * (t / TG)**exponent / np.exp(t / TG), text, second_ip_day,
+                   min_case_count)
+
+
+class EvaluatedFormula():
+    def __init__(self,
+                 formula,
+                 cumulative_active,
+                 first_idx,
+                 last_idx,
+                 first_date, xaxis_type):
+        self.text = formula.text
+        # First index: t=1
+        start_idx = np.argmax(cumulative_active >= formula.min_case_count)
+        length = last_idx - start_idx
+        self.y = formula.lambd(np.arange(length) + 1)
+        self.date_list = [(first_date + timedelta(days=d)).strftime('%Y-%m-%d')
+                          for d in range(start_idx, last_idx + 1)]
```

Comment:
For now it is fine. I think though that we can somehow refactor the formula concept such that it thas besides A and TG also a parameter T0. Then this shifts can be implemented in a more elegant way.

Reply:
Is `last_idx` supposed to be inclusive? Note that `len(self.date_list) = length + 1`.

Reply:
I think I had it consistent at some point but then change it back to how it looks in our current production. It doesn't matter much, whether we round up or down around the inflection point. At least I'll make it consistent now.

Reply:
Done